homeassistant/components/iotawatt/__init__.py

```diff
@@ -0,0 +1,149 @@
+"""The iotawatt integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from httpx import AsyncClient
+from iotawattpy.iotawatt import Iotawatt
+import voluptuous as vol
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_SCAN_INTERVAL
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import (
+    COORDINATOR,
+    DEFAULT_ICON,
+    DEFAULT_SCAN_INTERVAL,
+    DOMAIN,
+    IOTAWATT_API,
+    SIGNAL_ADD_DEVICE,
+)
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the iotawatt component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up iotawatt from a config entry."""
+    polling_interval = entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+
+    session = AsyncClient()
+    if "username" in entry.data.keys():
+        api = Iotawatt(
+            entry.data["name"],
+            entry.data["host"],
+            session,
+            entry.data["username"],
+            entry.data["password"],
+        )
+    else:
+        api = Iotawatt(
+            entry.data["name"],
+            entry.data["host"],
+            session,
+        )
+
+    coordinator = IotawattUpdater(
+        hass,
+        api=api,
+        name="IoTaWatt",
+        update_interval=polling_interval,
+    )
+
+    await coordinator.async_refresh()
+
+    if not coordinator.last_update_success:
+        raise ConfigEntryNotReady
+
+    hass.data[DOMAIN][entry.entry_id] = {
+        COORDINATOR: coordinator,
+        IOTAWATT_API: api,
```

Comment:
Why store the `api` object? It's already accessible as `coordinator.api`

Reply:
I removed the ```IOTAWATT_API: api,``` line of code, it's not needed.